client/src/views/common/BoostView.tsx

```diff
@@ -32,61 +41,86 @@ function onDeleteConfig(configType: ConfigT, name: string) {
   console.log(`name: ${name}`);
 }
 
-// Only dummy data
-const baseConfigs: BoostConfig[] = [
-  {
-    type: 'powerPlan',
-    options: [
-      { fileName: 'my_plan_1.json', displayName: 'my_plan_1' },
-      { fileName: 'this_one_gets_you_to_144.json', displayName: '144' },
-    ],
-    active: { fileName: 'my_plan_1.json', displayName: 'my_plan_1' },
-  },
-  {
-    type: 'rider',
-    options: [
-      { fileName: 'al.json', displayName: 'AL' },
-      { fileName: 'charles_rider.json', displayName: 'charles' },
-    ],
-    active: { fileName: 'charles_rider.json', displayName: 'charles' },
-  },
-  {
-    type: 'bike',
-    options: [
-      { fileName: 'blacksmith.json', displayName: 'Black Smith' },
-      { fileName: 'wombat.json', displayName: 'Wombat' },
-      { fileName: 'precilla.json', displayName: 'Precilla' },
-    ],
-    active: { fileName: 'wombat.json', displayName: 'Wombat' },
-  },
-  {
-    type: 'track',
-    options: [
-      { fileName: 'ford.json', displayName: 'Ford' },
-      { fileName: 'holden_track.json', displayName: 'Holden' },
-      { fileName: 'battle_mountain.json', displayName: 'Battle Mountain' },
-    ],
-    active: { fileName: 'ford.json', displayName: 'Ford' },
-  },
-];
-
 /**
  * Boost View component
  *
  * @returns {React.Component} Component
  */
 export default function BoostView() {
   // TODO: remove the hardcoded value for `distTravelled` with actual value read from MQTT
+  const [dist, setDist] = useState(0);
+  const [distOffset, setDistOffset] = useState<number | null>(null);
+  const [configs, setConfigs] = useState<BoostConfig[]>([
+    {
+      type: 'powerPlan',
+      options: [],
+    },
+    {
+      type: 'rider',
+      options: [],
+    },
+    {
+      type: 'bike',
+      options: [],
+    },
+    {
+      type: 'track',
+      options: [],
+    },
+  ]);
+
+  // fetch the reed distance from wireless module #3
+  const reedDistance = useSensorData(3, Sensor.ReedDistance, Number);
+
+  useEffect(() => {
+    if (reedDistance) {
+      // Set distance to 2 decimal places
+      setDist(Math.round(reedDistance * 100) / 100);
+    }
+  }, [reedDistance]);
```

Comment:
A couple of things regarding this part (and line 51).

The rounding here is premature as it is unnecessary and unused. `BoostCalibration` rounds the distance to 2dp anyway in `BoostCalibration.tsx` line 58 so rounding early actually makes it less accurate (possibly rounding inaccurately).

A stylistic point is that all of this code (including line 51) can be expressed succinctly in one line of code as `useSensorData` is already stateful. I found it a bit confusing to figure out what `dist` was supposed to be. 
If we want to have a default `0` value when the hook is `null`, we can use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) which essentially is "if null, set to this default value".

Reply:
`BoostCalibration` at present does not round the distance travelled, it rounds the calibrated distance. Nevertheless, I agree that the distance should be rounded where it's displayed, so in `BoostCalibration`, I'll fix that up.

In regards to the style issue, just to confirm, are you saying we don't need the `dist` state because `reedDistance` is basically acting like a state because `useSensorData` is stateful? And so would that mean we won't need the `useEffect` as well and we can just pass `reedDistance` to `BoostCalibartion`? Sorry if I'm misunderstanding it

Reply:
Yes that's correct. What we have now is
- `dist` is a duplicated state of `reedDistance`
- the `useEffect` rounds `reedDistance` and updates `dist` when `reedDistance` changes
So we could use only `reedDistance` here.